src/Ucommerce.Masterclass.Umbraco/Api/MasterClassBasketController.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Newtonsoft.Json;
+using MC_Headless.Exceptions;
+using MC_Headless.Headless;
+using Ucommerce.Masterclass.Umbraco.Models;
+using MC_Headless.Resolvers;
+using Umbraco.Web.WebApi;
+using Ucommerce.Headless.Domain;
+using Ucommerce;
+
+namespace MC_Headless.Api
+{
+    public class MasterClassBasketController : UmbracoApiController
+    {
+        private readonly ITransactionClient _transactionClient;
+        private readonly IPriceGroupIdResolver _priceGroupIdResolver;
+        private readonly ICultureCodeResolver _cultureCodeResolver;
+        private readonly IProductCatalogIdResolver _productCatalogIdResolver;
+
+        public MasterClassBasketController(ITransactionClient transactionClient, IPriceGroupIdResolver priceGroupIdResolver, ICultureCodeResolver cultureCodeResolver, IProductCatalogIdResolver productCatalogIdResolver)
+        {
+            _transactionClient = transactionClient;
+            _priceGroupIdResolver = priceGroupIdResolver;
+            _cultureCodeResolver = cultureCodeResolver;
+            _productCatalogIdResolver = productCatalogIdResolver;
+        }
+
+        public async Task<CheckoutViewModel> Get(CancellationToken ct)
+        {
+            return await GetCheckoutModel(ct);
+        }
+
+
+        [System.Web.Mvc.HttpPost]
+        public async Task<IHttpActionResult> UpdateOrderLine(UpdateOrderLineRequest updateOrderLineRequest, CancellationToken ct)
+        {
+            var basketId = this.Request.Headers.GetCookies().Select(c => c["basketId"])
```

Comment:
add null check and throw exception if basketId is null or whitespace. See GetCheckoutModel()